test/internal/tls.test.js

```diff
@@ -90,22 +90,23 @@ describe('trust-on-first-use', function() {
       fs.unlinkSync(knownHostsPath);
     }
 
-    fs.writeFileSync(
-      knownHostsPath,
-      'localhost:7687 abcd\n' +
-      'localhost:7687 abcd'
-    );
-
     driver = neo4j.driver("bolt://localhost", neo4j.auth.basic("neo4j", "neo4j"), {
       encrypted: true,
       trust: "TRUST_ON_FIRST_USE",
       knownHosts: knownHostsPath
     });
 
+    driver.session(); // write into the knownHost file
+
+    // duplicate the same serverId twice
+    setTimeout(function() {
+      var text = fs.readFileSync(knownHostsPath,'utf8');
+      fs.writeFileSync(knownHostsPath, text+text);
+    }, 1000);
+
     // When
     driver.session().run("RETURN true AS a").then( function(data) {
```

Comment:
Yes, but since `driver.session().run()` will fire off the queue at an arbitrary time, you cannot know if the file has been modified to have duplicated lines yet.